sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/http/HttpDownloadFileTask.java

```diff
@@ -0,0 +1,215 @@
+
+import android.net.TrafficStats;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+
+import com.microsoft.appcenter.http.TLS1_2SocketFactory;
+import com.microsoft.appcenter.utils.AppCenterLog;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.net.URLConnection;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import static com.microsoft.appcenter.distribute.DistributeConstants.LOG_TAG;
+import static com.microsoft.appcenter.http.HttpUtils.CONNECT_TIMEOUT;
+import static com.microsoft.appcenter.http.HttpUtils.READ_TIMEOUT;
+import static com.microsoft.appcenter.http.HttpUtils.THREAD_STATS_TAG;
+import static com.microsoft.appcenter.http.HttpUtils.WRITE_BUFFER_SIZE;
+
+/**
+ * Downloads an update and stores it on specified file.
+ **/
+class HttpDownloadFileTask extends AsyncTask<Void, Void, Void> {
+
+    /**
+     * Maximal number of allowed redirects.
+     */
+    private static final int MAX_REDIRECTS = 6;
+
+    /**
+     * The download progress will be reported after loading this number of bytes.
+     */
+    private static final long UPDATE_PROGRESS_BYTES_THRESHOLD = 128 * 1024;
+
+    /**
+     * The download progress will be reported not more often than this number of milliseconds.
+     */
+    private static final long UPDATE_PROGRESS_TIME_THRESHOLD = 200;
+
+    private static final String APK_CONTENT_TYPE = "application/vnd.android.package-archive";
+
+    private final HttpConnectionReleaseDownloader mDownloader;
+
+    /**
+     * The URI that hosts the binary to download.
+     */
+    private final Uri mDownloadUri;
+
+    /**
+     * The file that used to write downloaded package.
+     */
+    private final File mTargetFile;
+
+    HttpDownloadFileTask(HttpConnectionReleaseDownloader downloader, Uri downloadUri, File targetFile) {
+        mDownloader = downloader;
+        mDownloadUri = downloadUri;
+        mTargetFile = targetFile;
+    }
+
+    @Override
+    protected Void doInBackground(Void... args) {
+        try {
+            if (mTargetFile.exists()) {
+
+                /* Delete previously downloaded file if exists. */
+                if (!mTargetFile.delete()) {
+                    AppCenterLog.warn(LOG_TAG, "Cannot delete previously downloaded file: " + mTargetFile);
+                }
+            }
+
+            /* Create connection. */
+            URL url = new URL(mDownloadUri.toString());
+            TrafficStats.setThreadStatsTag(THREAD_STATS_TAG);
+            URLConnection connection = createConnection(url, MAX_REDIRECTS);
+            connection.connect();
+
+            /* Content type check. Produce only warning if it doesn't match. */
+            String contentType = connection.getContentType();
+            if (contentType == null || !contentType.equals(APK_CONTENT_TYPE)) {
```

Comment:
```suggestion
            if (!APK_CONTENT_TYPE.equals(contentType)) {
```
![yoda condition](https://cdn.images.express.co.uk/img/dynamic/36/590x/Star-Wars-What-killed-Yoda-837201.jpg)